src/main/java/traincraft/blocks/trainworkbench/ContainerTrainWorkbench.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Traincraft
+ * Copyright (c) 2011-2020.
+ *
+ * This file ("ContainerTrainWorkbench.java") is part of the Traincraft mod for Minecraft.
+ * It is created by all people that are listed with @author below.
+ * It is distributed under LGPL-v3.0.
+ * You can find the source code at https://github.com/Traincraft/Traincraft
+ */
+
+package traincraft.blocks.trainworkbench;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.*;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.play.server.SPacketSetSlot;
+import net.minecraft.world.World;
+import net.minecraftforge.event.entity.living.LivingEquipmentChangeEvent;
+import org.apache.logging.log4j.Level;
+import traincraft.Traincraft;
+import traincraft.api.SlotCraftingResult;
+
+import java.util.ArrayList;
+
+/**
+ * @author PseudonymPatel
+ * @since 2020-7-30
+ */
+// NOTE: basically using the exact same system as the vanilla workbench
+public class ContainerTrainWorkbench extends Container {
+
+    public static ArrayList<TrainWorkbenchRecipe> TRAINWORKBENCH_RECIPES = new ArrayList<>();
```

Comment:
Shouldn't this be inside the TileEntity class?

Reply:
I put it there because it's there where the recipe handling happens. I thought it would make more sense to put it with where the handling happens, but I can change it if you would like.

Reply:
you also can put it into the Recipe class itself, this makes the most sense

Reply:
I moved it into TrainWorkbenchRecipe